handler/book.go

```diff
@@ -0,0 +1,68 @@
+package handler
+
+import (
+	"encoding/json"
+	"github.com/ryota1116/stacked_books/domain/model/googleBooksApi"
+	"github.com/ryota1116/stacked_books/handler/http/request"
+	"github.com/ryota1116/stacked_books/usecase"
+	"io/ioutil"
+	"net/http"
+)
+
+type BookHandlerInterface interface {
+	SearchBooks(w http.ResponseWriter, r *http.Request)
+}
+
+type bookHandler struct {
+	bookUseCase usecase.BookUseCaseInterface
+}
+
+func NewBookHandler(bu usecase.BookUseCaseInterface) BookHandlerInterface {
+	return &bookHandler{
+		bookUseCase: bu,
+	}
+}
+
+// SearchBooks : 外部APIを用いた書籍検索のエンドポイント
+func (bh bookHandler) SearchBooks(w http.ResponseWriter, r *http.Request)  {
+	// 
+	var requestParameter googleBooksApi.RequestParameter
```

Comment:
handlerは、ユースケースがどういうAPIを利用しているか知らないはずなので、
「GoogleBooksApi」のバリデーションではなく、「（船先さんが作った）書籍検索API」のバリデーションを責務に持った方が良いと思いました！

```suggestion
	var requestParameter searchBooksApi.RequestParameter
```

Reply:
確かにです🙇‍♂️ 71ff24d でHandlerに移しました！
命名は `handler/http/request/book/search_books_request.go`にしてみました👀